src/Microsoft.Extensions.Primitives/StringValues.cs

```diff
@@ -253,6 +254,256 @@ public static StringValues Concat(StringValues values1, StringValues values2)
             return new StringValues(combined);
         }
 
+        public static bool Equals(StringValues left, StringValues right)
+        {
+            if (IsNullOrEmpty(left))
+            {
+                return IsNullOrEmpty(right);
+            }
+
+            if (left._values == null)
```

Comment:
It seems like this could be dramatically simplified by using the counts + indexer - any reason why you didn't do it this way?